### ISSUE TITLE
General plan outline

### ISSUE THREAD

**Issue by msalvaris** [Azure/fast_retraining#18]
Speeding up Machine Learning Applications with LightGBM library in Real Time domains

Short description
The speed of a machine learning algorithm can be crucial in problems that requires retraining in real time. Some useful domains can be IoT, sport result prediction, predictive maintenance and healthcare. Microsoft recently open sourced LightGBM library for decision trees that outperformed other libraries in both speed and performance. In this talk we will demo several applications using LightGBM.

Abstract
In some applications training and retraining times need to be kept below 5 seconds to be useful. Such applications are often referred to as real-time and include but are not limited to IoT, sport result prediction, predictive maintenance and healthcare. Algorithms that allow for fast retraining are fundamental to enabling such applications and can open-up new business opportunities. One reason for retraining is that the features used in these applications can degrade causing previously useful features to no longer be useful. Such degradation is often observed as sensors age or as information becomes out-of-date.
LightGBM is a new open source library created by Microsoft that is set to become the new standard in decision trees algorithms. Depending on the application, it can be anything from 4 to 10 times faster than XGBoost and offers a higher accuracy. It has already been proven useful in several Kaggle competitions.
In this talk we will explore this promising library, compare it with the current state of the art and demo a business case of a real-time application.

Slides
1) Concept drift (5 minutes)
* Introduction
* Intro to Concept Drift
* What ways can we combat concept Drift
2) XGBoost & LightGBM (10-15 minutes)
* Intro to the unreassonble effectivness of XGBoost
* What is LightGBM
* Summary of XGBoost vs LightGBM training speed, execution speed and accuracy
    * Planet Kaggle
    * Flights
    * ...
* Why is LightGBM faster
3) Real time applications (10 minutes)
* Real time application
	* BCI
	* Airplane
	* IoT?
	* ...
* Business case?
